### PR TITLE
Add searching

### PR DESCRIPTION
**Pull request by aletor123**

### CODE REVIEW

File: apps/vanilla/src/components/AnimeTableComponents/filterPanel.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    searchFilter.placeholder = 'Type to search';
+    searchFilterContainer.append(
+      searchFilter,
+    );
+    this.searchFilter = searchFilter;
+    const searchFilterChangedEvent$ = fromEvent(searchFilter, 'input');
+    const searchFilterChangedEventResult$ = searchFilterChangedEvent$.pipe(
+      debounceTime(1000),
+    );
+    searchFilterChangedEventResult$.subscribe(() => this.handleSearchChanged());
```
Comment (timofei-maximov-saritasa) [inline apps/vanilla/src/components/AnimeTableComponents/filterPanel.ts:64]: That's great that you decided to use rxjs, but let's use vanilla solution. Also don't forget to unsubscribe :)

File: apps/vanilla/src/components/AnimeTableComponents/animeTable.ts [hunk 2]
```diff
@@ -150,6 +171,13 @@ export interface PaginationRequestParams {
   offset: number;
 }
 
+/** Filters request params. */
+export interface FiltersParams {
```
Comment (kristinaSvobodinaSaritasa) [inline apps/vanilla/src/components/AnimeTableComponents/animeTable.ts:175]: I think `FilterParams` (without s) will be better.

File: package.json [hunk 3]
```diff
@@ -29,9 +29,11 @@
     "@angular/platform-browser-dynamic": "~14.0.0",
     "@angular/router": "~14.0.0",
     "@reduxjs/toolkit": "^1.8.2",
+    "@types/lodash.debounce": "^4.0.7",
```
Comment (timofei-maximov-saritasa) [inline package.json:32]: This should go to `devDependencies`

**Review by timofei-maximov-saritasa — APPROVED** [review]

File: apps/vanilla/src/components/AnimeTableComponents/animeTable.ts [hunk 4]
```diff
@@ -62,11 +69,21 @@ export class AnimeTable {
...
+
   private paginationPanel: PaginationPanel = new PaginationPanel({
     updateMethod: this.updatePaginationState,
     defaultPaginationParams: this.state.paginationParams,
   });
 
+  private filterPanel: FilterPanel = new FilterPanel({
+    updateMethod: this.updateFiltersState,
+    defaultFilterParams: this.state.filterParams,
+  });
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/components/AnimeTableComponents/animeTable.ts:85]: You can omit type annotation here.

File: apps/vanilla/src/components/AnimeTableComponents/animeTable.ts [hunk 5]
```diff
@@ -150,6 +171,13 @@ export interface PaginationRequestParams {
   offset: number;
 }
 
+/** Filters request params. */
+export interface FilterParams {
+
+  /** Search term. */
+  searchTerm: string;
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/components/AnimeTableComponents/animeTable.ts:178]: don't forget readonly.

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

File: apps/vanilla/src/components/AnimeTableComponents/filterPanel.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
+import debounce from 'lodash.debounce';
+
+import { filtersStyles } from '../../constants/styles/animeTable';
+
+import {
+  FilterParams,
+} from './animeTable';
+
+const SEARCH_DEBOUNCE_TIME = 1000;
```
Comment (chernodub) [inline apps/vanilla/src/components/AnimeTableComponents/filterPanel.ts:9]: Usually we prefer using a suffix representing time units, this improves code readability. `SEARCH_DEBOUNCE_MS` for your case

**Review by chernodub — APPROVED** [review]

**Pull request closed without merging**